src/bpftrace.cpp

```diff
@@ -361,46 +394,9 @@ void perf_event_printer(void *cb_cookie, void *data, int size __attribute__((unu
     auto args = std::get<1>(bpftrace->system_args_[id]);
     auto arg_values = bpftrace->get_arg_values(args, arg_data);
 
-    char buffer [255];
-
-    switch (args.size())
-    {
-      case 0:
-        ret = system(fmt);
-        break;
-      case 1:
-        snprintf(buffer, 255, fmt, arg_values.at(0)->value());
-        ret = system(buffer);
-        break;
-      case 2:
-        snprintf(buffer, 255, fmt, arg_values.at(0)->value(), arg_values.at(1)->value());
-        ret = system(buffer);
-        break;
-      case 3:
-        snprintf(buffer, 255, fmt, arg_values.at(0)->value(), arg_values.at(1)->value(), arg_values.at(2)->value());
-        ret = system(buffer);
-        break;
-      case 4:
-        snprintf(buffer, 255, fmt, arg_values.at(0)->value(), arg_values.at(1)->value(), arg_values.at(2)->value(),
-          arg_values.at(3)->value());
-        ret = system(buffer);
-        break;
-      case 5:
-        snprintf(buffer, 255, fmt, arg_values.at(0)->value(), arg_values.at(1)->value(), arg_values.at(2)->value(),
-          arg_values.at(3)->value(), arg_values.at(4)->value());
-        ret = system(buffer);
-        break;
-      case 6:
-        snprintf(buffer, 255, fmt, arg_values.at(0)->value(), arg_values.at(1)->value(), arg_values.at(2)->value(),
-          arg_values.at(3)->value(), arg_values.at(4)->value(), arg_values.at(5)->value());
-        ret = system(buffer);
-        break;
-      default:
-        std::cerr << "printf() can only take up to 7 arguments (" << args.size() << ") provided" << std::endl;
-        abort();
-    }
-    (void)ret;
-
+    char buffer[255];
+    format(buffer, 255, fmt, arg_values);
```

Comment:
I'm just thinking - would there be any problem with longer system calls like with printf?

Even if we do want to keep it limited, we shouldn't run the command if our string is truncated. Could get some nasty things happening.

Reply:
I couldn't find a reason but just left it as is.

Will do this:
```
int size = format(buffer, BUFSIZE, fmt, arg_values);
// Return value is required size excluding null byte
if (size >= BUFSIZE) {
  std::cerr << "syscall() command to long" << std::endl;
  abort();
}
```